### PR TITLE
fix(post-processing): prefer LICENSE over NOTICE when co-hosted legal files disagree

### PR DESCRIPTION
**Pull request by mstykow**
## Summary

- When a package's co-located legal files disagree, ADR 0010's `promote_package_declared_license_from_legal_files` abstained and left `declared_license_expression` null. The ubiquitous Apache-style `LICENSE` + `NOTICE` pattern hit this: a canonical `LICENSE` states the package's own license while the `NOTICE` enumerates *bundled third-party* licenses, so the two disagree.
- This PR prefers the canonically-named license-family files (`LICENSE`/`LICENCE`/`COPYING` and their `LICENSE-*`/`COPYING-*` variants) over `NOTICE`/`COPYRIGHT`/`AUTHORS` on disagreement, and promotes from that subset. A `NOTICE` listing bundled-dependency licenses no longer drives or blocks the package's declared license.
- Makes Provenant **more correct than ScanCode**, which over-aggregates both files into a noisy `apache-2.0 AND bsd-new AND mit`.

## Scope and exclusions

- Included: new `is_license_family_file` classifier; LICENSE-over-NOTICE precedence in the ADR 0010 promotion pass; unit tests; ADR 0010 amendment documenting the precedence + rationale.
- Explicit exclusions: no change to file-level `package_data` stamping (still rejected per ADR 0010); all other ADR 0010 guards (genuine-absence, same-dir/sole-package, provenance, single/agreeing-file, resolved-dependency) are unchanged. The narrowing applies **only on disagreement and only when a canonical license file is present**, so agreeing sets and `NOTICE`/`COPYING`-only directories keep prior behavior.

## How to verify

- Scan a repo with a root `LICENSE` (the project license) and a root `NOTICE` listing bundled third-party licenses (e.g. `prometheus/prometheus@351447a4`) with `--package --license`. Confirm the assembled `pkg:golang/github.com/prometheus/prometheus` now reports `declared_license_expression: apache-2.0` sourced only from `LICENSE`, where it was previously `null`. Confirmed locally on that ref.
- Edge cases covered by unit tests in `package_metadata_promotion.rs`: dual `LICENSE-APACHE` + `LICENSE-MIT` still abstains even with a co-located agreeing `NOTICE`; a `NOTICE`-only directory still promotes from `NOTICE`.

## Intentional differences from Python

- ScanCode over-aggregates `LICENSE` + `NOTICE` into a combined expression; Provenant promotes only the package's own `LICENSE` license and treats `NOTICE` as bundled-attribution. Documented in ADR 0010 "LICENSE-over-NOTICE precedence".

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/adr/0010-package-license-from-cohosted-files.md [hunk 1]
```diff
@@ -254,6 +275,58 @@ assembled-`Package` pass, and the file-level remainder is all false-positive ris
...
+
+- **Dual-license `LICENSE-*` files still abstain.** The license-family subset can
+  itself be genuinely ambiguous — `LICENSE-APACHE` (`apache-2.0`) + `LICENSE-MIT`
+  (`mit`) are both canonical license files and they disagree. The single-file /
+  agreeing-files test runs _within_ the chosen subset, so this dual-license case is
+  unchanged and still left unset. A co-located `NOTICE` does not break the tie.
+- **`NOTICE`/`COPYING`-only directories are unchanged.** If a directory has _no_
+  canonical license-family file, the pass falls back to the full legal set, so a sole
+  `NOTICE` (or `COPYRIGHT`/`AUTHORS`) still promotes exactly as before. The precedence
+  only re-prioritizes; it never removes a previously-available signal.
```
[the remark this answers is not in the document]
Comment (mstykow) [reply docs/adr/0010-package-license-from-cohosted-files.md]: Fixed in 0f263187: the bullet header now reads `NOTICE`/`COPYRIGHT`-only and explicitly notes that a `COPYING`-only directory takes the license-family path, not the fallback. Good catch.

File: src/post_processing/package_metadata_promotion.rs [hunk 2]
```diff
@@ -148,21 +158,47 @@ pub(super) fn promote_package_declared_license_from_legal_files(
...
+            }
+        };
+
         // The legal files' agreed expression carries the authoritative operator
         // structure — crucially an `OR` choice, which must not be tightened to `AND`.
-        // Abstain when separate legal files disagree (e.g. dual `LICENSE-APACHE` +
-        // `LICENSE-MIT`).
+        // Abstain when separate legal files (within the chosen subset) still disagree
+        // (e.g. dual `LICENSE-APACHE` + `LICENSE-MIT`).
         let Some(file_expression) = single_declared_expression(&source_legal_files) else {
```
[the remark this answers is not in the document]
Comment (mstykow) [reply src/post_processing/package_metadata_promotion.rs]: Addressed in 0f263187: extracted the selection into `select_legal_files_and_expression`, which returns both the chosen file set and its agreed expression in one pass. The agreeing path now captures the expression directly instead of re-running `single_declared_expression`; the second resolution only happens on the narrowed subset in the disagreement path, where it is genuinely needed.

**Comment by mstykow** [comment]
Addressed the "Comments Outside Diff" finding (missing agreeing `LICENSE` + `NOTICE` test) in 0f263187: added `promotes_when_license_and_notice_agree`, which exercises the `Some(_)` (agreeing) arm of the selection for a mixed license-family + attribution set and asserts both detections are kept with no narrowing.
